sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_common/_base_eval.py

```diff
@@ -0,0 +1,273 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from typing import Optional, List, Dict, Callable, Any
+import inspect
+
+from abc import ABC
+
+from typing import Dict, List
+
+import numpy as np
+from promptflow._utils.async_utils import async_run_allowing_running_loop
+
+from azure.ai.evaluation._exceptions import EvaluationException, ErrorBlame, ErrorCategory, ErrorTarget
+
+
+# TODO exception target pass down?
+class _BaseEval(ABC):
```

Comment:
nit: as we are not exposing those base classes, why not just remove underscore prefix?

Reply:
Also, if you don't mind, can we rename

_BaseEval -> EvaluatorBase
_BasePromptyEval -> PromptyEvaluatorBase
_BaseRaiServiceEval -> RaiServiceEvaluatorBase

Reply:
Done on the next commit in 10 minutes.